app/feature/feature-payments/src/main/kotlin/com/hedvig/android/feature/payments2/navigation/PaymentsGraph.kt

```diff
@@ -36,7 +37,9 @@ fun NavGraphBuilder.paymentsGraph(
           navigator.navigateUnsafe(PaymentsDestinations2.History(paymentOverview))
         },
         onChangeBankAccount = navigateToConnectPayment,
-        onDiscountClicked = { }, // TODO
+        onDiscountClicked = { discounts ->
+          navigator.navigateUnsafe(PaymentsDestinations2.Discounts(discounts))
```

Comment:
The `unsafe` variant exists for scenarios where we do not want to drop the event in case there is an already ongoing navigation going on.
To give examples:

We usually use the `safe` alternative, so that if there's an ongoing navigation, we simply do not do anything. This is important particularly when we know that this can be triggered more than once very quickly for some reason.

If we want to navigate as a response to an event coming from the ViewModel, where we know this will only happen once, we are okay with potentially doing `navigateUnsafe` since we want this navigation to happen even if they for example have a bottom sheet showing, which would put the lifecycle state in Paused, which would mean the `navigate` function would not perform the navigation. Because that's the difference between the normal one and the `unsafe` one. The normal one checks if the lifecycle state is just resumed (aka nothing is going on, no navigation, no bottom sheet is showing, the app is not on the background, etc.), while unsafe just navigated regardless.

However if this even happens on a click of a button, as we do here, and then we do `navigateUnsafe`, it means that if someone clicks this button twice really fast, we will navigate there two times actually. And pressing back would bring you back to the same screen and then you'd have to press back again.

Reply:
Awesome, thanks